spot_driver_cpp/include/spot_driver_cpp/images/spot_image_publisher_node.hpp

```diff
@@ -4,16 +4,22 @@
 
 #include <rclcpp/node_interfaces/node_base_interface.hpp>
 #include <rclcpp/node_options.hpp>
-#include <spot_driver_cpp/spot_image_publisher.hpp>
+#include <spot_driver_cpp/api/spot_api.hpp>
+#include <spot_driver_cpp/images/spot_image_publisher.hpp>
+#include <spot_driver_cpp/interfaces/logger_interface_base.hpp>
+#include <spot_driver_cpp/interfaces/parameter_interface_base.hpp>
 
 #include <memory>
 
-namespace spot_ros2 {
+namespace spot_ros2::images {
 /**
  * @brief Wraps SpotImagePublisher to allow using it like a rclcpp::Node.
  */
 class SpotImagePublisherNode {
  public:
+  SpotImagePublisherNode(std::shared_ptr<rclcpp::Node> node, std::unique_ptr<SpotApi> spot_api,
+                         const std::shared_ptr<ParameterInterfaceBase>& parameter_interface,
+                         const std::shared_ptr<LoggerInterfaceBase>& logger_interface);
```

Comment:
I think there needs to be a version of this constructor that takes a `SpotImagePublisher::MiddlewareHandle` as an argument to allow us to instantiate this class in tests without requiring a real instance of `rclcpp::Node`. This version of the constructor can create the ImagesMiddlewareHandle and then call the other constructor.

Reply:
It seems like there are only two cases we need to support here:
- We have a real Node, and we are creating rclcpp-specialized versions of the middleware handle and the other interface classes.
- We are in a unit test and we are using mock or fake versions of the middleware handle and interface classes.
So there isn't a situation where we need to mix having a real Node and fake/mock interface classes.

With that in mind, you can probably replace this constructor withe one like this:

```
SpotImagePublisherNode(std::unique_ptr<SpotImagePublisher::MiddlewareHandle> middleware_handle, std::unique_ptr<SpotApi> spot_api,
                         const std::shared_ptr<ParameterInterfaceBase>& parameter_interface,
                         const std::shared_ptr<LoggerInterfaceBase>& logger_interface)
```